src/goose/cli/session.py

```diff
@@ -156,10 +156,10 @@ def run(self) -> None:
                 self.reply()  # Process the user message.
             except KeyboardInterrupt:
                 self.interrupt_reply()
-            except Exception:
+            except FailedToGenerateMessageError:
+                # rewind to right before the last user message
+                self.exchange.rewind_to_last_user_message()
                 print(traceback.format_exc())
-                if self.exchange.messages:
-                    self.exchange.messages.pop()
                 print(
                     "\n[red]The error above was an exception we were not able to handle.\n\n[/]"
```

Comment:
Maybe update this print statement to say a bit more clear about clearing the state to the just before the previous user prompt?